exes/ethercat/inc/public/tfc/cia/402.hpp

```diff
@@ -28,6 +28,25 @@ struct control_word {
   bool reserved_4 : 1 {};
   bool reserved_5 : 1 {};
 
+  static constexpr auto from_uint(std::uint16_t word) noexcept -> control_word {
+    return control_word{ .operating_state_switch_on = static_cast<bool>(word & 0b1),
+                         .enable_voltage = static_cast<bool>(word & 0b10),
+                         .operating_state_quick_stop = static_cast<bool>(word & 0b100),
+                         .enable_operation = static_cast<bool>(word & 0b1000),
+                         .operating_mode_specific_0 = static_cast<bool>(word & 0b10000),
+                         .operating_mode_specific_1 = static_cast<bool>(word & 0b100000),
+                         .operating_mode_specific_2 = static_cast<bool>(word & 0b1000000),
+                         .fault_reset = static_cast<bool>(word & 0b10000000),
+                         .halt = static_cast<bool>(word & 0b100000000),
+                         .operating_mode_specific_3 = static_cast<bool>(word & 0b1000000000),
+                         .reserved_0 = static_cast<bool>(word & 0b10000000000),
+                         .reserved_1 = static_cast<bool>(word & 0b100000000000),
+                         .reserved_2 = static_cast<bool>(word & 0b1000000000000),
+                         .reserved_3 = static_cast<bool>(word & 0b10000000000000),
+                         .reserved_4 = static_cast<bool>(word & 0b100000000000000),
+                         .reserved_5 = static_cast<bool>(word & 0b1000000000000000) };
```

Comment:
```suggestion
    return std::bit_cast<control_word>(word);
```

Reply:
NICE !

Reply:
But this is not implemented with libc++ as constexpr, as can be seen in pipelines.
Artur has implementation that I would like to integrate to https://github.com/arturbac/small_vectors/blob/master/include/utils/utility_cxx20.h#L198
I would like to officially finish integrating this repo(small_vectors) with vcpkg as dependency so we can utilize this and updated version of fixed_string, @omarhogni any counters?

Reply:
This does not make sense to me https://reviews.llvm.org/rGb1fb3d75c953fa2e02ebddb6ebbf100f99786f0c bit_cast is constexpr under libc++ and was added in c++20 as constexpr. Are you sure there is not another reason?

Reply:
they have tests for this error being thrown https://github.com/llvm/llvm-project/blob/2e6f722b88e42bba72af5bafd8cc1e37e460d978/clang/test/SemaCXX/constexpr-builtin-bit-cast.cpp#L143

I can disable the tests on clang

Reply:
> But this is not implemented with libc++ as constexpr, as can be seen in pipelines. Artur has implementation that I would like to integrate to https://github.com/arturbac/small_vectors/blob/master/include/utils/utility_cxx20.h#L198 I would like to officially finish integrating this repo(small_vectors) with vcpkg as dependency so we can utilize this and updated version of fixed_string, @omarhogni any counters?

I don't think this will be constexpr under clang. We are stuck for now. This is more complex than using another implementation see notes under https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2018/p0476r2.html  especially
```
Furthermore, it is currently impossible to implement a constexpr bit-cast function, as memcpy itself isn’t 
constexpr. Marking the proposed function as constexpr doesn’t require or prevent memcpy from becoming 
constexpr, but requires compiler support.
```

Reply:
disabled on clang